Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -1194,10 +1194,18 @@ static Ref<CSSValue> valueForGridTrackList(GridTrackSizingDirection direction, R
     bool isRowAxis = direction == GridTrackSizingDirection::ForColumns;
     auto* renderGrid = dynamicDowncast<RenderGrid>(renderer);
     bool isSubgrid = isRowAxis ? style.gridSubgridColumns() : style.gridSubgridRows();
-    bool isMasonry = (direction == GridTrackSizingDirection::ForRows) ? style.gridMasonryRows() : style.gridMasonryColumns();
     auto& trackSizes = isRowAxis ? style.gridColumnTrackSizes() : style.gridRowTrackSizes();
     auto& autoRepeatTrackSizes = isRowAxis ? style.gridAutoRepeatColumns() : style.gridAutoRepeatRows();
 
+    // If masonry is specified for both grid-template-columns and grid-template-rows,
+    // then the used value for grid-template-columns is 'none'.
+    if (style.gridMasonryRows() && style.gridMasonryColumns())
+        return direction == GridTrackSizingDirection::ForColumns ? CSSPrimitiveValue::create(CSSValueNone) : CSSPrimitiveValue::create(CSSValueMasonry);
```

Comment:
This is just changing the computed value, not the used value

Reply:
I misread the spec here. Fixed the code now.